### DISCUSSION TITLE
Appending fields in a complex structure

### DISCUSSION THREAD

**Discussion by hfrankel** [react-hook-form/react-hook-form#6848] [Q&A]
Hi there,

Just wondering if I can please get please get help on implementing dynamic fields in a complex form structure?

I am working on a wizard form at the moment which should have the ability for the user to dynamically add and remove some of the fields on some steps of the form. I've been trying to work with useFieldArray and am struggling to implement it properly for the project requirements.

Ideal data structure sent to the backend would be:
`{
 "firstName": "Test",
 "lastName": "User",
 "favouriteSports": [
  { "favouriteSport": "Tennis" },
  { "favouriteSport": "Cricket" }
 ],
"favouriteFoods": [
{ "favouriteFood": "Ice cream" },
  { "favouriteFood": "Pizza" }
]
}`

Or for a larger fieldset something like:
`{
 "name": "Test User",
 "carWishList": [
  { "make": "Tesla", "model": "Cyber Truck" "year": "2020" }
 ],
 "birthdayWishList": [
  { "item": "new phone" }
  ]
}`

Does anyone have any recommendations or guidance on a best approach to this? Any help is much appreciated! Thanks

**Comment by bluebill1049** [comment]
are you using `useFieldArray`? maybe share a codesandbox as well.

**Reply by hfrankel** [reply]
Yeah I'm using `useFieldArray`. I'll build a codesandbox example then share.

**Reply by hfrankel** [reply]
@bluebill1049 got a basic [example](https://codesandbox.io/s/usefieldarray-example-c9xwy?file=/src/App.js) working in the end.

Next stage of the implementation will be to map through the ArrayFields component. If you have any suggestions on improvements that would still be very helpful.
